presto-main/src/main/java/com/facebook/presto/sql/analyzer/GroupingOperationAnalyzer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.sql.tree.DefaultExpressionTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.GroupingOperation;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+public class GroupingOperationAnalyzer
```

Comment:
Take a look at: `com.facebook.presto.sql.testing.TreeAssertions#linearizeTree`. I think you could have similar approach (override `process` method).